collector/src/toolchain.rs

```diff
@@ -288,6 +292,45 @@ impl ToolchainComponents {
     }
 }
 
+#[derive(Clone, Copy)]
+pub struct ToolchainConfig<'a> {
+    rustdoc: Option<&'a Path>,
+    clippy: Option<&'a Path>,
+    cargo: Option<&'a Path>,
+    id: Option<&'a str>,
+}
+
+impl<'a> ToolchainConfig<'a> {
+    pub fn default() -> Self {
```

Comment:
Somewhat ironically, clippy tells you to derive Default for this struct :laughing: https://github.com/rust-lang/rustc-perf/actions/runs/6273536371/job/17037178205?pr=1724

Reply:
Done :cat:!